nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -182,12 +182,14 @@ class BtActionNode : public BT::ActionNodeBase
       // user defined callback. May modify the value of "goal_updated_"
       on_wait_for_result();
 
-      auto goal_status = goal_handle_->get_status();
-      if (goal_updated_ && (goal_status == action_msgs::msg::GoalStatus::STATUS_EXECUTING ||
-        goal_status == action_msgs::msg::GoalStatus::STATUS_ACCEPTED))
-      {
-        goal_updated_ = false;
-        on_new_goal_received();
+      if (goal_handle_) {
```

Comment:
If this condition is not true (e.g. the `goal_handle_` is not valid), then we pass over this which seems reasonable since we have no valid status. However, then we just continue the looping process. Is the idea that the result callback will be triggered sometime in the near-term future to set `result_` so that this BT node would then exit?

Reply:
I did not think about this case.
Is following snippet ok?
```
      if (goal_handle_) {
        auto goal_status = goal_handle_->get_status();
        if (goal_updated_ && (goal_status == action_msgs::msg::GoalStatus::STATUS_EXECUTING ||
          goal_status == action_msgs::msg::GoalStatus::STATUS_ACCEPTED))
        {
          goal_updated_ = false;
          on_new_goal_received();
        }

         rclcpp::spin_some(node_);

         // check if, after invoking spin_some(), we finally received the result
         if (!goal_result_available_) {
           // Yield this Action, returning RUNNING
           return BT::NodeStatus::RUNNING;
         }
      }
```
this snippet is not tested yet.

Reply:
I don't think this solves the problem -- `if goal_handle_ == false`, then the node that's being `ticked` will never exit, there should be some exit criteria. This is what both of these comments are pointing to, there's an issue where if not goal_handle_, then the BT node would never exit the running state. One or the other here needs a way to handle the case (probably the other makes more sense) so that it can exit.